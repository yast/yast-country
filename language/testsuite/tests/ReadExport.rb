# encoding: utf-8

# tests for Language.ycp autoyast functions: prepare for cloning
# Author:	jsuchome@suse.cz
# $Id$
module Yast
  class ReadExportClient < Client
    def main
      # testedfiles: Language.ycp

      Yast.import "Testsuite"

      @READ = {
        "sysconfig" => {
          "language" => {
            "RC_LANG"             => "de_DE.UTF-8",
            "INSTALLED_LANGUAGES" => "en_US,de_DE"
          }
        },
        "target"    => {
          "size"  => 1,
          "dir"   => ["language_de_DE.ycp"],
          "yast2" => {
            "de_DE" => ["Deutsch", "Deutsch", ".UTF-8", "@euro", "German"],
            "en_US" => ["English", "English", ".UTF-8", "@euro", "English"]
          }
        }
      }

      @E = { "target" => { "bash_output" => {} } }

      Yast.import "Mode"
      Mode.SetMode("autoinst_config")

      Testsuite.Init([@READ, {}, @E], nil)

      Yast.import "Language"

      Testsuite.Test(lambda { Language.Read(true) }, [@READ, {}, @E], nil)

      Testsuite.Test(lambda { Language.Export }, [], nil)

      Testsuite.Test(lambda { Language.Summary }, [], nil)

      Testsuite.Test(lambda { Language.Modified }, [], nil)

      nil
    end
  end
end

Yast::ReadExportClient.new.main
