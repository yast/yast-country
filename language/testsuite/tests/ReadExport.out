Read	.target.dir "/usr/share/YaST2/data/languages" ["language_de_DE.ycp"]
Read	.target.yast2 "languages/language_de_DE.ycp" $["de_DE":["Deutsch", "Deutsch", ".UTF-8", "@euro", "German"], "en_US":["English", "English", ".UTF-8", "@euro", "English"]]
Return	true
Return	$["language":"en_US", "languages":"en_US,de_DE"]
Return	<ul><li>Primary Language: English</li><li>Additional Languages: <ul><li>German</li></ul></li></ul>
Return	false
